twilight-lavalink/src/model/incoming.rs

```diff
@@ -0,0 +1,363 @@
+        Self::Ready(event)
+    }
+}
+
+impl From<Event> for IncomingEvent {
+    fn from(event: Event) -> IncomingEvent {
+        Self::Event(event)
+    }
+}
+
+impl From<PlayerUpdate> for IncomingEvent {
+    fn from(event: PlayerUpdate) -> IncomingEvent {
+        Self::PlayerUpdate(event)
+    }
+}
+
+impl From<Stats> for IncomingEvent {
+    fn from(event: Stats) -> IncomingEvent {
+        Self::Stats(event)
+    }
+}
+
+/// The discord voice information that lavalink uses for connection and sending information.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[non_exhaustive]
+#[serde(rename_all = "camelCase")]
+pub struct VoiceState {
+    /// The Discord voice token to authenticate with.
+    pub token: String,
+    /// The Discord voice endpoint to connect to.
+    pub endpoint: String,
+    /// The Discord voice session id to authenticate with. Note this is separate from the lavalink session id.
+    pub session_id: String,
+}
+
+/// An update about the information of a player. Filters are currently unsupported
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[non_exhaustive]
+#[serde(rename_all = "camelCase")]
+pub struct PlayerUpdate {
+    /// Op code for this websocket event.
+    pub op: Opcode,
+    /// The guild ID of the player.
+    pub guild_id: Id<GuildMarker>,
+    /// The new state of the player.
+    pub state: PlayerUpdateState,
+}
+
+/// New statistics about a node and its host.
+#[derive(Clone, Debug, Deserialize, Eq, PartialEq, Serialize)]
+#[non_exhaustive]
+#[serde(rename_all = "camelCase")]
+pub struct PlayerUpdateState {
+    /// Unix timestamp of the player in milliseconds.
+    pub time: i64,
+    /// Track position in milliseconds. None if not playing anything.
+    pub position: i64,
+    /// True when the player is connected to the voice gateway.
+    pub connected: bool,
+    /// The ping of the node to the Discord voice server in milliseconds (-1 if not connected).
+    pub ping: i64,
+}
+
+/// Dispatched by Lavalink upon successful connection and authorization. Contains fields determining if resuming was successful, as well as the session id.
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+#[non_exhaustive]
+#[serde(rename_all = "camelCase")]
+pub struct Ready {
+    /// Op code for this websocket event.
+    pub op: Opcode,
+    /// Whether this session was resumed.
+    pub resumed: bool,
+    /// The Lavalink session id of this connection. Not to be confused with a Discord voice session id.
+    pub session_id: String,
+}
+
+/// Statistics about a node and its host.
+#[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+#[non_exhaustive]
+#[serde(rename_all = "camelCase")]
+pub struct Stats {
+    /// Op code for this websocket event.
+    pub op: Opcode,
+    /// CPU information about the node's host.
+    pub cpu: StatsCpu,
+    /// Statistics about audio frames.
+    #[serde(rename = "frameStats", skip_serializing_if = "Option::is_none")]
```

Comment:
Is this rename necessary with the previous camel case rename?

Reply:
I don't think the rename is anymore. Didn't catch that before. Not sure the reason for that previously. I'll update that. The serializing might also be able to be dropped now in v4. I'll double check.

Reply:
That is dropped now. Was not needed.